tests/suites/static_analysis/lint_shell.sh

```diff
@@ -20,7 +21,8 @@ run_whitespace() {
 }
 
 run_trailing_whitespace() {
-  OUT=$(grep -nr " $" tests/ | grep '\.sh:' || true)
+  # shellcheck disable=SC2063
+  OUT=$(grep -n -r --include "*.sh" " $" tests/ | grep -oP "^.*:\d+" || true)
```

Comment:
What does the ':' do in the last regex?

Reply:
So we end up with the `filename.sh:<linenumber>:filename.sh<error>`. What we want to tell the user is `filename.sh:<linenumber>`. The duplicated `filename.sh<error>` is the problem, we want to tell grep to just show matches of that.

Ignore everything else.